frontend/src/metabase/querying/components/FilterModal/SegmentFilterEditor/SegmentFilterEditor.tsx

```diff
@@ -13,17 +12,14 @@ export function SegmentFilterEditor({
   segmentItems,
   onChange,
 }: SegmentFilterEditorProps) {
-  const data = useMemo(() => {
-    return segmentItems.map((segmentItem, segmentIndex) => ({
-      value: String(segmentIndex),
-      label: segmentItem.displayName,
-      isSelected: segmentItem.filterPositions.length > 0,
```

Comment:
There was a React warning for `isSelect` prop being applied to the DOM element, so I'm splitting `data` into `options` and the actual `data` passed to `MultiSelect`